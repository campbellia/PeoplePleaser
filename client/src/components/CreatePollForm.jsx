import React, { useState } from 'react';

const CreatePollForm = (props) => {

  const [name, setName] = useState(null);
  const [options, setOptions] = useState([]);
  const [newOption, setNew] = useState(null);

  const handleAddOption = (e) => {
    e.preventDefault();
    setOptions(options.concat(newOption));
  }

  const getOptionsList = () => {
    if (options.length) {
      return options.map(option => {
        return <div>{option}</div>;
      });
    } else {
      return <div>none</div>;
    }
  }

  const onFormSubmit = (e) => {
    e.preventDefault();
    var optionsObj = {};
    options.forEach(option => {
      optionsObj[option] = 0;
    });
    props.onFormSubmit(name, optionsObj);
  }

  return (
    <div>
      <form onSubmit={(e) => {onFormSubmit(e)}}>
        Name:<input type="text" onChange={(e) => {setName(e.target.value)}}></input>
        <input id="createPoll" type="submit" value="New Poll"></input>
      </form>
        Options: {getOptionsList()}
      <form onSubmit={(e) => {handleAddOption(e)}}>
        New Option: <input type="text" onChange={(e) => {setNew(e.target.value)}}></input>
        <input id="addOption" type="submit" value="Add"></input>
      </form>
    </div>
  );
}

export default CreatePollForm;