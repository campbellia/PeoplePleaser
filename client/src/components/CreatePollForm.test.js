import React from 'react';
import { shallow } from 'enzyme';
import CreatePollForm from './CreatePollForm';

describe('CreatePollForm', () => {
  it('should render correctly', () => {
    const component = shallow(<CreatePollForm />);
    expect(component).toMatchSnapshot();
  });

  it('should allow the user to add new options to the poll', () => {
    const component = mount(<CreatePollForm />);
    component.state('newOption') = 'Bananas';
    component
      .find('.addOption')
      .simulate('click');

    expect(component.state('options').length).toEqual(1);
    expect(component.state('options')[0]).toEqual('Bananas');

    component.state('newOption') = 'Crabapples';
    component
      .find('.addOption')
      .simulate('click');
    expect(component.state('options').length).toEqual(2);
    expect(component.state('options').includes('Bananas')).toEqual(true);
    expect(component.state('options').includes('Crabapples')).toEqual(true);

    component.unmount();
  });

});

//should be able to add options
//should call event handler when submit
//should change state when options added
//