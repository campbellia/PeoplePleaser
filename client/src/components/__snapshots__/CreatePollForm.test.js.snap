// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePollForm should render correctly in "debug" mode 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    Name:
    <input
      onChange={[Function]}
      type="text"
    />
    <input
      type="submit"
      value="New Poll"
    />
  </form>
  Options: 
  <div>
    none
  </div>
  <form
    onSubmit={[Function]}
  >
    New Option: 
    <input
      onChange={[Function]}
      type="text"
    />
    <input
      type="submit"
      value="Add"
    />
  </form>
</div>
`;
